kaiax/gasless/impl/tx_pool.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2024 The Kaia Authors
+// This file is part of the Kaia library.
+//
+// The Kaia library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Kaia library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Kaia library. If not, see <http://www.gnu.org/licenses/>.
+
+package impl
+
+import (
+	"math"
+
+	"github.com/kaiachain/kaia/blockchain/types"
+	"github.com/kaiachain/kaia/kaiax"
+)
+
+func (g *GaslessModule) PreAddLocal(tx *types.Transaction) error {
+	return nil
+}
+
+func (g *GaslessModule) PreAddRemote(tx *types.Transaction) error {
+	return nil
+}
+
+func (g *GaslessModule) IsModuleTx(_ kaiax.TxPool, tx *types.Transaction) bool {
+	return g.IsApproveTx(tx) || g.IsSwapTx(tx)
+}
+
+func (g *GaslessModule) GetCheckBalance() func(pool kaiax.TxPool, tx *types.Transaction) error {
+	return func(kaiax.TxPool, *types.Transaction) error { return nil }
+}
```

Comment:
why return a callback instead of checking balance here?
If needed, we can add statedb to GaslessModule, which is also required for nonce check (SP3).

Reply:
If the balance check is overwritten simply, all modules need to implement this even if they want to use the existing logic. So I provided the option to overwrite logic.